### PR TITLE
[Refactor] Remove boost filesystem

### PR DESCRIPTION
**Pull request by yangzhg**
## Proposed changes

use std::filesystem instead of boost

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [x] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have created an issue on (#5575) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/util/filesystem_util.cc [hunk 1]
```diff
@@ -21,28 +21,25 @@
...
 namespace doris {
 
 Status FileSystemUtil::create_directory(const string& directory) {
     error_code errcode;
-    bool exists = filesystem::exists(directory, errcode);
+    bool exists = std::filesystem::exists(directory, errcode);
     // Need to check for no_such_file_or_directory error case - Boost's exists() sometimes
     // returns an error when it should simply return false.
-    if (errcode != errc::success && errcode != errc::no_such_file_or_directory) {
+    if (errcode&& errcode != std::errc::no_such_file_or_directory) {
```
Comment (stdpain) [inline be/src/util/filesystem_util.cc:42]: need a code format here

**Comment by stdpain** [comment]
May be CMakeList.txt Boost::filesystem could be removed

File: be/test/olap/olap_snapshot_converter_test.cpp [hunk 2]
```diff
@@ -17,13 +17,13 @@
 
 #include "olap/olap_snapshot_converter.h"
 
-#include <boost/algorithm/string.hpp>
+#include <boost/algorithm/string.hpp>\
```
Comment (morningman) [inline be/test/olap/olap_snapshot_converter_test.cpp:20]:
```suggestion
#include <boost/algorithm/string.hpp>
```

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**